lib/spack/spack/util/compression.py

```diff
@@ -312,37 +300,23 @@ def decompressor_for(path, extension=None):
         return decompressor_for_nix(extension)
 
 
-def decompressor_for_nix(extension):
-    """Returns a function pointer to appropriate decompression
-    algorithm based on extension type and unix specific considerations
-    i.e. a reasonable expectation system utils like gzip, bzip2, and xz are
-    available
+def decompressor_for_nix(extension: str) -> Callable[[str], str]:
+    """Returns a function pointer to appropriate decompression algorithm based on extension type
+    and unix specific considerations i.e. a reasonable expectation system utils like gzip, bzip2,
+    and xz are available
 
     Args:
-        path (str): path of the archive file requiring decompression
+        extension: path of the archive file requiring decompression
     """
-    if re.match(r"zip$", extension):
-        return _unzip
-
-    if re.match(r"gz$", extension):
-        return _gunzip
-
-    if re.match(r"bz2$", extension):
-        return _bunzip2
+    extension_to_decompressor: Dict[str, Callable[[str], str]] = {
+        "zip": _unzip,
```

Comment:
@johnwparent I guess the regexes were a leftover from a time where you passed a path or file name? The regexes are an involved way of doing `==`.

Reply:
Those regexes have been a part of Spack longer than I have lol.
Prior to Windows development, when this module was ~60 LOC, the only logic in `decompressor_for` was a bunch of regex checks like this, I just folded them into the updated logic:
https://github.com/spack/spack/blob/v0.15.0/lib/spack/spack/util/compression.py#L25

Reply:
Yeah okay, the code you reference takes a path not an extension, so after a refactor they became redundant.

Reply:
I'm not sure what you're driving at here, I'm not disagreeing with you, they should come out.

FWIW the code referenced takes both a path and extension, and the regexs are made only against the extension, which is still effectively `==` (except slightly fuzzier w.r.t nested extensions).

Reply:
I meant to say that the code is confusing, either cause I was missing something, or because of an oversight in refactor / review earlier. Was looking for confirmation it was the former, which I got. All good.